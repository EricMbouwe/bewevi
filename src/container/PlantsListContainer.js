import { useSelector } from 'react-redux';
import PlantList from '../component/PlantList';

const PlantListContainer = () => {
  const plants = useSelector(state => state.plant.plants);
  return (

    <PlantList plants={plants} />
  );
};

export default PlantListContainer;
