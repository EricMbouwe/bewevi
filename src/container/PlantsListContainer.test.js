import React from 'react';
import { render, screen } from '@testing-library/react';
import { useSelector } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import PlantsListContainer from './PlantsListContainer';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

beforeEach(() => {
  useSelector.mockImplementation(selector => selector(
    {
      plant: {
        plants: [{
          id: 1, common_name: 'Ivy common', scientific_name: 'Ivy scientific', image_url: 'Ivy.jpg',
        }, {
          id: 2, common_name: 'Ivy common2', scientific_name: 'Ivy scientific2', image_url: 'Ivy2.jpg',
        }],
      },
    },
  ));
});

describe('<PlantsListContainer />', () => {
  it('is rendering PlantsList component', () => {
    render(
      <BrowserRouter>
        <PlantsListContainer />
      </BrowserRouter>,
    );
    expect(screen.getAllByText(/Ivy scientific/i).length).toEqual(2);
  });

  it('renders correctly', () => {
    const renderedContainer = render(
      <BrowserRouter>
        <PlantsListContainer />
      </BrowserRouter>,
    );
    expect(renderedContainer).toMatchSnapshot();
  });
});
