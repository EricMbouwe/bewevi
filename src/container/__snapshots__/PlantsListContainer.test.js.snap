// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlantsListContainer /> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="plants-list main"
      >
        <a
          class="plant-short main"
          href="/one-plant"
        >
          <div
            class="plant-short description"
          >
            <div>
              Ivy common
            </div>
            <div>
              Ivy scientific
            </div>
          </div>
          <div
            class="plant-short image-container"
          >
            <img
              alt="plant"
              class="plant-short plant-image"
              src="Ivy.jpg"
            />
          </div>
        </a>
        <a
          class="plant-short main"
          href="/one-plant"
        >
          <div
            class="plant-short description"
          >
            <div>
              Ivy common2
            </div>
            <div>
              Ivy scientific2
            </div>
          </div>
          <div
            class="plant-short image-container"
          >
            <img
              alt="plant"
              class="plant-short plant-image"
              src="Ivy2.jpg"
            />
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="plants-list main"
    >
      <a
        class="plant-short main"
        href="/one-plant"
      >
        <div
          class="plant-short description"
        >
          <div>
            Ivy common
          </div>
          <div>
            Ivy scientific
          </div>
        </div>
        <div
          class="plant-short image-container"
        >
          <img
            alt="plant"
            class="plant-short plant-image"
            src="Ivy.jpg"
          />
        </div>
      </a>
      <a
        class="plant-short main"
        href="/one-plant"
      >
        <div
          class="plant-short description"
        >
          <div>
            Ivy common2
          </div>
          <div>
            Ivy scientific2
          </div>
        </div>
        <div
          class="plant-short image-container"
        >
          <img
            alt="plant"
            class="plant-short plant-image"
            src="Ivy2.jpg"
          />
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
